### ISSUE TITLE
Executable project is not considered for serialization type resolution

### ISSUE THREAD

**Issue by manio143** [stride3d/stride#879]
**Release Type**: Official Release/GitHub (either)

**Version**: master

**Platform(s)**: N/A

**Describe the bug**
In the `Stride.Core.Assets.Package` we only consider library assemblies to be loaded. This means if I have a script in my executable it would not be resolved for deserialization.

**To Reproduce**
1. Create an empty .NET Console application (with a solution)
2. Include Stride packages (`Stride.Engine` and `Stride.Core.Assets.CompilerApp`)
3. Create Assets folder in the project directory
4. Run GameStudio on the solution to add in the GameSettings, MainScene, GraphicsCompositor
5. Create a Script in the project

It's not possible to add the script to the scene in GameStudio. If added "by force" to the MainScene file, the AssetCompiler will warn that YamlSerializer couldn't deserialize it.

**Expected behavior**
It should be possible to create a single assembly game and have the same behavior as with library projects.

**Additional context**
I have modified the `Package.cs` on my dev branch to fix it:
https://github.com/manio143/stride/blob/f67805d7c6a00fb5a57dd17f6b8e46933ffc9d2e/sources/assets/Stride.Core.Assets/Package.cs#L1038-L1054
I will observe for a while if there's any issues and if not I'll submit a proper PR

**Comment by manio143** [comment]
Solves  #833 _Script Editor bug_

**Comment by manio143** [comment]
Note: mobile platforms (UWP, Android, iOS) need to be excluded here, because they don't have a proper assembly output (exe/dll) and instead are packaged during compilation.

**Comment by xen2** [comment]
Yes, the reason default is to separate library from executable is that executable projects is often a `TargetFramework` incompatible for loading within Game Studio (i.e. .NET Core project with .NET FW game studio).
However, it's true that if the user works only with a specific platform, the game studio can load, it's not a problem to do it.
Not sure how to properly expose that in a user-friendly way...
Maybe have a template for a "simple desktop game" that contains only an executable project matching the current framework of running editor (for now it could be either .NET Core or .NET FW, but soon it should be even more simple with only one option).
